### PR TITLE
[FEMDEM] Adding a consistent strategy for the 2way coupling of the FEM-DEM

### PR DESCRIPTION
**Pull request by AlejandroCornejo**
In this PR I'm adding a new strategy for the FEM-DEM in which the DEM calculate contact forces at each iteration (in the former version only computed at the end of the step) so now increases the non-linearity but is more exact

### CODE REVIEW

**Comment by philbucher** [comment]
FYI @KlausBSautter this seems similar to what you are doing => maybe you can take a look at how the strong coupling is done here

File: applications/FemToDemApplication/python_scripts/MainCoupling2WayFemDem.py [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+import KratosMultiphysics.MeshingApplication.mmg_process as MMG
+import KratosMultiphysics.DEMApplication as KratosDEM
+import KratosMultiphysics.DemStructuresCouplingApplication as DemFem
+import KratosMultiphysics.FemToDemApplication.MainCouplingFemDem as MainCouplingFemDem
+
+def Wait():
+    input("Press Something")
+
+#============================================================================================================================
+class MainCoupled2WayFemDem_Solution(MainCouplingFemDem.MainCoupledFemDem_Solution):
```
Comment (maceligueta) [inline applications/FemToDemApplication/python_scripts/MainCoupling2WayFemDem.py:19]: Hi @AlejandroCornejo . I strongly recommend to atomize the methods in the class `MainCouplingFemDem.MainCoupledFemDem_Solution`
I would take pieces of code which can be encapsulated in functions and move them to different methods. Then, when you derive, like you are doing here, you will duplicate a lot less lines.

Comment (AlejandroCornejo) [reply applications/FemToDemApplication/python_scripts/MainCoupling2WayFemDem.py:19]: You're totally right. I'll do another PR cleaning everything

**Comment by maceligueta** [comment]
You could also address Codacy's suggestions, they are quite easy to fix.  I just saw that it's saying that Initialize is too complex XD, well that's for a different PR. Tell me when you need final approval.

**Comment by AlejandroCornejo** [comment]
> 
> 
> You could also address Codacy's suggestions, they are quite easy to fix. I just saw that it's saying that Initialize is too complex XD, well that's for a different PR. Tell me when you need final approval.

Now I've corrected everything and done the codacy suggestions @maceligueta . Feeel free to approve THX

**Review by maceligueta — APPROVED** [review]

**Pull request merged**